openedx/core/djangoapps/content/course_overviews/signals.py

```diff
@@ -112,18 +162,60 @@ def _check_for_pacing_changes(previous_course_overview, updated_course_overview)
 
 
 def _check_for_cert_availability_date_changes(previous_course_overview, updated_course_overview):
-    """ Checks if the cert available date has changed and if so, sends a COURSE_CERT_DATE_CHANGE signal"""
-    if previous_course_overview.certificate_available_date != updated_course_overview.certificate_available_date:
+    """
+    Checks if the certificate available date (CAD) or the certificates display behavior (CDB) of a course run has
+    changed during a course overview update. If so, we emit the COURSE_CERT_DATE_CHANGE signal to ensure other parts of
+    the system are aware of the change. The `credentials` app listens for this signal in order to keep our certificate
+    visibility settings in the Credentials IDA up to date.
+
+    Args:
+        previous_course_overview (CourseOverview): the current course overview instance for a particular course run
+        updated_course_overview (CourseOverview): an updated course overview instance, reflecting the current state of
+            data from the modulestore/Mongo
+
+    Returns:
+        None
+    """
+    def _send_course_cert_date_change_signal():
```

Comment:
[QUESTION] I know this isn't your code, but do you know why it's structured this way? Can `transaction.on_commit()` not just take the `COURSE_CERT_DATE_CHANGE.send_robust()` call directly?

Reply:
IIRC, this is related to a bugfix made by a former Aperturian. I tried looking up the original commit that added this line and Livvy left an incredible amount of detail in the commit: https://github.com/openedx/edx-platform/commit/74780ad4c059113d340c13ddb965e3b248e9ad2c.

Reply:
Beyond that, it looks like `on_commit` expects a function/callable, so I don't think we can just pass the call directly? I did assume this was setup this way for a reason.